fennel/client/client.py

```diff
@@ -403,10 +403,44 @@ def extract_features(
                 )
         return df
 
-    def extract_historical_features(
+    def extract_features(
         self,
-        input_feature_list: List[Union[Feature, Featureset, str]],
+        input_feature_list: List[Union[Feature, str]],
         output_feature_list: List[Union[Feature, Featureset, str]],
+        input_dataframe: pd.DataFrame,
+        log: bool = False,
+        workflow: Optional[str] = None,
+        sampling_rate: Optional[float] = None,
+    ) -> Union[pd.DataFrame, pd.Series]:
+        """
+        Going to be deprecated in favor of extract and will be removed in the future.
+        Extract features for a given output feature list from an input
+        feature list. The features are computed for the current time.
+
+        Parameters:
+        input_feature_list (List[Union[Feature, str]]): List of feature objects or fully qualified feature names (when providing a str) can be used as input. We don't allow adding featureset as input because if an engineer adds a new feature to the featureset it would break all extract calls running in production.
+        output_feature_list (List[Union[Feature, Featureset, str]]): List of feature objects or featuresets or fully qualified feature names (when providing a str) to compute.
+        input_dataframe (pd.DataFrame): Dataframe containing the input features.
+        log (bool): Boolean which indicates if the extracted features should also be logged (for log-and-wait approach to training data generation). Default is False.
+        workflow (Optional[str]): The name of the workflow associated with the feature extraction. Only relevant when log is set to True.
+        sampling_rate (float): The rate at which feature data should be sampled before logging. Only relevant when log is set to True. The default value is 1.0.
+
+        Returns:
+        Union[pd.DataFrame, pd.Series]: Pandas dataframe or series containing the output features.
+        """
+        return self.extract(
```

Comment:
Can you log a warning regarding this deprecation, same for extract_historical_features

Reply:
Added